### PR TITLE
regenerate gradle lockfiles & roll pub packages

### LINKED ISSUE #145561: Make `generate_gradle_lockfiles.dart` script check exit code of `flutter build apk --config-only`
It currently does not verify that the command succeeds: https://github.com/flutter/flutter/blob/deaa600204d8d561f705d8adbaf031f16ec3fc80/dev/tools/bin/generate_gradle_lockfiles.dart#L102.

The script should fail if the command fails

### PR DESCRIPTION
**Pull request by christopherfujino**
Example failing devicelab build https://ci.chromium.org/ui/p/flutter/builders/prod/Linux_pixel_7pro%20platform_views_scroll_perf__timeline_summary/2980/overview

Fixes https://github.com/flutter/flutter/issues/145561

### CODE REVIEW

**Comment by christopherfujino** [comment]
This previously landed in https://github.com/flutter/flutter/pull/145509 but was reverted in https://github.com/flutter/flutter/pull/145550

File: dev/benchmarks/test_apps/stocks/lib/i18n/stock_strings.dart [hunk 1]
```diff
@@ -14,14 +14,14 @@ import 'package:intl/intl.dart' as intl;
...
 
-/// Callers can lookup localized strings with an instance of StockStrings returned
-/// by `StockStrings.of(context)`.
+/// Callers can lookup localized strings with an instance of StockStrings
+/// returned by `StockStrings.of(context)`.
 ///
 /// Applications need to include `StockStrings.delegate()` in their app's
-/// localizationDelegates list, and the locales they support in the app's
-/// supportedLocales list. For example:
+/// `localizationDelegates` list, and the locales they support in the app's
```
Comment (christopherfujino) [inline dev/benchmarks/test_apps/stocks/lib/i18n/stock_strings.dart:21]: @gmackall Does the regenerate gradle lockfiles script change dartdocs?

Comment (christopherfujino) [reply dev/benchmarks/test_apps/stocks/lib/i18n/stock_strings.dart:21]: Hmm, these diffs also broke analysis:

```
Analyzing flutter...                                            

   info • The expression has no effect and can be removed • dev/benchmarks/test_apps/stocks/lib/i18n/stock_strings.dart:69:83 • noop_primitive_operations
   info • Convert 'locale' to a super parameter • dev/benchmarks/test_apps/stocks/lib/i18n/stock_strings_en.dart:12:3 • use_super_parameters
   info • Convert 'locale' to a super parameter • dev/benchmarks/test_apps/stocks/lib/i18n/stock_strings_es.dart:12:3 • use_super_parameters
```

Comment (christopherfujino) [reply dev/benchmarks/test_apps/stocks/lib/i18n/stock_strings.dart:21]: This happened before: https://github.com/flutter/flutter/pull/137190/files#r1371888460

It's unclear to me how that script would touch dart files, though...

Comment (christopherfujino) [reply dev/benchmarks/test_apps/stocks/lib/i18n/stock_strings.dart:21]: Oh, we're calling `flutter pub get`, which WILL do localization code generation

**Comment by gmackall** [comment]
The `generate_gradle_lockfiles.dart` script is poorly named, as it also re-generates all the gradle files. That is going to make one of the tests here fail `deprecated_gradle_settings_test`. 

I think the answer isn't to change the test but to add a flag to the script to allow running it in a mode where we don't re-generate the gradle files themselves (I don't see a reason why the pub roller bot should need to re generate the gradle files themselves for each roll)

**Review by gmackall — COMMENTED** [review]
Looks like the `gradle_deprecated_settings` test is still failing. I think the issue here is that `generate gradle files` -> `generate lockfiles` -> `revert grade file generation` is not exactly the same as just `generate lockfiles`.

I was able to reproduce the error with those steps, but it built successfully if I instead started on a new branch and just did `flutter update-packages --force-upgrade` followed by running the lockfile generation script with the `--no-gradle-generation` flag

**Comment by christopherfujino** [comment]
> Looks like the `gradle_deprecated_settings` test is still failing. I think the issue here is that `generate gradle files` -> `generate lockfiles` -> `revert grade file generation` is not exactly the same as just `generate lockfiles`.
> 
> I was able to reproduce the error with those steps, but it built successfully if I instead started on a new branch and just did `flutter update-packages --force-upgrade` followed by running the lockfile generation script with the `--no-gradle-generation` flag

Thanks

**Comment by christopherfujino** [comment]
Superceded by: https://github.com/flutter/flutter/pull/145727

**Pull request closed without merging**